### PR TITLE
[Compute] Fix #22995: `az image-version create`: Unbind the usage of `--target-region-encryption` and `--target-region-cvm-encryption`

### LINKED ISSUE #22995: Cannot create CVM SIG Image Version via CLI
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Related command**
az sig image version create

**Describe the bug**
Cannot create CVM sig image versions from CLI.

**To Reproduce**
Create a sig image definition with confidential vm security type.  If using CLI, here's an example: 
`az sig image-definition create --resource-group tiregan --gallery-name tireganCvmGallery --gallery-image-definition ConfidentialVmImage --publisher tiregan --offer tiregan --sku tiregan --os-type windows --os-state Generalized --features SecurityType=ConfidentialVM --hyper-v-generation V2`

Then, create a sig image version under this image.  It requires target-region-cvm-encryption parameter.  example:
`az sig image-version create --resource-group NEURG --gallery-name CVMGallery --gallery-image-definition CVMWinGenImage --gallery-image-version 1.0.0 --os-snapshot WinCVMEncrypteddisk --target-region-cvm-encryption EncryptedWithPmk --target-regions northeurope`

The above fails.  The error returned is this:
`(InvalidParameter) The regional encryption.osDiskImage.securityProfile.type must be specified when the SecurityType of the image definition is ConfidentialVM.
Code: InvalidParameter
Message: The regional encryption.osDiskImage.securityProfile.type must be specified when the SecurityType of the image definition is ConfidentialVM.
Target: galleryArtifactVersion.properties.publishingProfile.targetRegions.encryption.osDiskImage.securityProfile`

**Expected behavior**
The above should successfully create a SIG image version with proper CVM properties.

The request that comes to the backend SIG service should look similar to this:
```
{
  "location": "northeurope",
  "tags": {},
  "properties": {
    "publishingProfile": {
      "targetRegions": [
        {
          "name": "northeurope",
          "encryption": {
            "osDiskImage": {
              "securityProfile": {
                "confidentialVMEncryptionType": "EncryptedWithPmk"
              }
            }
          }
        }
      ]
    },
    "storageProfile": {
      "osDiskImage": {
        "source": {
          "id": "/subscriptions/5fb7f578-4dfb-476a-b46b-52696f8ecc32/resourceGroups/NEURG/providers/Microsoft.Compute/snapshots/WinCVMEncrypteddisk"
        }
      }
    }
  }
}
```

But instead the request comes in missing the encryption.osDiskImage.securityProfile and so the request fails.

**Environment summary**
N/A

**Additional context**
It’s worth noting that I did successfully make a CVM image version via Azure CLI.  However, in order to do that I had to also specify the –target-region-encryption parameter.  The target-region-encryption and the target-region-cvm-encryption parameters should be independent of one another but right now the target-region-cvm-encryption parameter ONLY works when -target-region-encryption is also specified.
As an example of this, this command actually worked:
`                az sig image-version create --resource-group tiregan --gallery-name tireganCvmGallery --gallery-image-definition ConfidentialVmImage --gallery-image-version 1.0.2 --os-snapshot [source snapshot id] --target-region-cvm-encryption EncryptedWithPmk, --target-regions eastus2euap --target-region-encryption [des id]`

See email thread with subject 'RE: Problem creating Confidential Gallery Image Version via CLI'

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Related command**

The `--target-region-encryption` and the `--target-region-cvm-encryption` parameters should be independent of one another but right now the `--target-region-cvm-encryption` parameter ONLY works when `--target-region-encryption` is also specified.
https://github.com/Azure/azure-cli/issues/22995

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
compute

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -2052,58 +2052,63 @@ def process_gallery_image_version_namespace(cmd, namespace):
...
-                os_disk_image = terms[0]
-                if os_disk_image == 'null':
-                    os_disk_image = None
-                else:
-                    des_id = _disk_encryption_set_format(cmd, namespace, os_disk_image)
-                    security_profile = None
-                    if hasattr(namespace, 'target_region_cvm_encryption') and namespace.target_region_cvm_encryption:
-                        cvm_terms = namespace.target_region_cvm_encryption[i].split(',')
-                        if not cvm_terms or len(cvm_terms) != 2:
+            if hasattr(namespace, 'target_region_encryption') or hasattr(namespace, 'target_region_cvm_encryption'):
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:2055]: Please decouple those judgment logic to reduce the complexity of code logic

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -2052,58 +2052,63 @@ def process_gallery_image_version_namespace(cmd, namespace):
...
+                                                                  secure_vm_disk_encryption_set_id=cvm_des_id)
+                    if os_disk_image:
+                        os_disk_image.security_profile = security_profile
+                    else:
+                        os_disk_image = OSDiskImageEncryption(security_profile=security_profile)
+                encryption = EncryptionImages()
+                if os_disk_image:
+                    encryption.os_disk_image = os_disk_image
+                if data_disk_images:
+                    encryption.data_disk_images = data_disk_images
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:2111]:
```suggestion
                encryption = EncryptionImages(os_disk_image=os_disk_image, data_disk_images=data_disk_images)
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 3]
```diff
@@ -2052,58 +2052,63 @@ def process_gallery_image_version_namespace(cmd, namespace):
...
-                # OS disk
-                os_disk_image = terms[0]
-                if os_disk_image == 'null':
-                    os_disk_image = None
-                else:
-                    des_id = _disk_encryption_set_format(cmd, namespace, os_disk_image)
-                    security_profile = None
-                    if hasattr(namespace, 'target_region_cvm_encryption') and namespace.target_region_cvm_encryption:
-                        cvm_terms = namespace.target_region_cvm_encryption[i].split(',')
-                        if not cvm_terms or len(cvm_terms) != 2:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:2066]: Please keep the original logic

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 4]
```diff
@@ -2100,10 +2080,31 @@ def process_gallery_image_version_namespace(cmd, namespace):
...
+                    cvm_des_id = _disk_encryption_set_format(cmd, namespace, cvm_terms[1])
+                security_profile = OSDiskImageSecurityProfile(confidential_vm_encryption_type=cvm_terms[0],
+                                                              secure_vm_disk_encryption_set_id=cvm_des_id)
+                if os_disk_image:
+                    os_disk_image.security_profile = security_profile
                 else:
-                    data_disk_images = None
+                    os_disk_image = OSDiskImageEncryption(security_profile=security_profile)
+            if os_disk_image or data_disk_images:
                 encryption = EncryptionImages(os_disk_image=os_disk_image, data_disk_images=data_disk_images)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:2107]:
```suggestion

            if os_disk_image or data_disk_images:
                encryption = EncryptionImages(os_disk_image=os_disk_image, data_disk_images=data_disk_images)

```

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 5]
```diff
@@ -2100,8 +2080,31 @@ def process_gallery_image_version_namespace(cmd, namespace):
...
                             lun=lun, disk_encryption_set_id=des_id))
                     data_disk_images = data_disk_image_encryption_list
+
+            if hasattr(namespace, 'target_region_cvm_encryption') and namespace.target_region_cvm_encryption:
+                cvm_terms = namespace.target_region_cvm_encryption[i].split(',')
+                if not cvm_terms or len(cvm_terms) != 2:
+                    raise ArgumentUsageError(
+                        "usage error: {} is an invalid target region cvm encryption. "
+                        "Both os_cvm_encryption_type and os_cvm_des parameters are required.".format(cvm_terms))
+                storage_profile_types = [profile_type.value for profile_type in ConfidentialVMEncryptionType]
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:2090]:
```suggestion
                        "Both os_cvm_encryption_type and os_cvm_des parameters are required.".format(cvm_terms))

                storage_profile_types = [profile_type.value for profile_type in ConfidentialVMEncryptionType]
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 6]
```diff
@@ -2061,29 +2063,7 @@ def process_gallery_image_version_namespace(cmd, namespace):
                 else:
                     des_id = _disk_encryption_set_format(cmd, namespace, os_disk_image)
                     security_profile = None
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:2065]:
```suggestion
```

**Pull request merged**